src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Activity.cs

```diff
@@ -128,5 +128,23 @@ public async Task ReindexInstancesAsync([ActivityTrigger] ReindexInstanceInput i
 
             await Task.WhenAll(tasks);
         }
+
+        /// <summary>
+        ///  Fetch schema version.
+        /// </summary>
+        /// <param name="context">The durable activity context.</param>
+        /// <param name="log">The log</param>
+        /// <returns>The task.</returns>
+        [FunctionName(nameof(FetchSchemaVersionAsync))]
+        public async Task FetchSchemaVersionAsync([ActivityTrigger] IDurableActivityContext context, ILogger log)
+        {
+            // TODO: performance improvement, don't need to call service for every call.
+            EnsureArg.IsNotNull(context, nameof(context));
+            EnsureArg.IsNotNull(log, nameof(log));
+            log.LogInformation("Fetching schema version");
+            int version = await _schemaManagerDataStore.GetCurrentSchemaVersionAsync(default);
```

Comment:
This is odd. This is done as part of SchemaInitializer.Start, we should not be doing this here

Reply:
Will could explain more on this. :)